GPflowOpt/acquisition/mes.py

```diff
@@ -0,0 +1,94 @@
+# Copyright 2017 Joachim van der Herten, Nicolas Knudde
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from .acquisition import Acquisition
+from ..design import RandomDesign
+
+from GPflow import settings
+from GPflow.param import DataHolder
+
+import numpy as np
+from scipy.stats import norm
+from scipy.optimize import bisect
+import tensorflow as tf
+
+float_type = settings.dtypes.float_type
+stability = settings.numerics.jitter_level
+
+
+class MinValueEntropySearch(Acquisition):
+    """
+        Max-value entropy search acquisition function for single-objective global optimization.
+        Introduced by (Wang et al., 2017).
+
+        Key reference:
+
+        ::
+            @InProceedings{Wang:2017,
+              title = 	 {Max-value Entropy Search for Efficient {B}ayesian Optimization},
+              author = 	 {Zi Wang and Stefanie Jegelka},
+              booktitle = 	 {Proceedings of the 34th International Conference on Machine Learning},
+              pages = 	 {3627--3635},
+              year = 	 {2017},
+              editor = 	 {Doina Precup and Yee Whye Teh},
+              volume = 	 {70},
+              series = 	 {Proceedings of Machine Learning Research},
+              address = 	 {International Convention Centre, Sydney, Australia},
+              month = 	 {06--11 Aug},
+              publisher = 	 {PMLR},
+            }
+        """
+
+    def __init__(self, model, domain, gridsize=10000, num_samples=10):
+        super(MinValueEntropySearch, self).__init__(model)
+        self.gridsize = gridsize
+        self.num_samples = num_samples
+        self.samples = DataHolder(np.zeros(num_samples))
+        self._domain = domain
+        self.setup()
+
+    def setup(self):
+        super(MinValueEntropySearch, self).setup()
+        m = self.models[0]
+        X = m.X.value
+        Xrand = RandomDesign(self.gridsize, self._domain).generate()
```

Comment:
I only scanned the paper quickly last week. Does it make sense to accept a design directly in the CTor instead of gridsize and the domain?

At the very least we can create the RandomDesign in the CTor? and just call self.design.generate()